WalletWasabi.Fluent/ViewModels/AddWallet/Create/ConfirmRecoveryWordsViewModel.cs

```diff
@@ -9,63 +8,155 @@
-#endif
-		var nextCommandCanExecute =
-			_confirmationWordsSourceList
+		_availableWords = new List<RecoveryWordViewModel>();
+		_words = words;
+		_mnemonic = mnemonic;
+		_walletName = walletName;
+	}
+
+	public ObservableCollectionExtended<RecoveryWordViewModel> ConfirmationWords { get; } = new();
+
+	[System.Diagnostics.CodeAnalysis.SuppressMessage("Reliability", "CA2000:Dispose objects before losing scope", Justification = "Uses DisposeWith()")]
+	protected override void OnNavigatedTo(bool isInHistory, CompositeDisposable disposables)
+	{
+		base.OnNavigatedTo(isInHistory, disposables);
+
+		ConfirmationWords.Clear();
+
+		var confirmationWordsSourceList = new SourceList<RecoveryWordViewModel>();
+
+		confirmationWordsSourceList
+			.DisposeWith(disposables)
 			.Connect()
 			.ObserveOn(RxApp.MainThreadScheduler)
-			.WhenValueChanged(x => x.IsConfirmed)
-			.Select(_ => _confirmationWordsSourceList.Items.All(x => x.IsConfirmed));
+			.Sort(SortExpressionComparer<RecoveryWordViewModel>.Ascending(x => x.Index))
+			.Bind(ConfirmationWords)
+			.OnItemAdded(x => x.Reset())
+			.Subscribe()
+			.DisposeWith(disposables);
 
 		EnableBack = true;
 
-		NextCommand = ReactiveCommand.CreateFromTask(() => OnNextAsync(mnemonic, walletName), nextCommandCanExecute);
+		CancelCommand = ReactiveCommand.Create(OnCancel);
 
-		if (_isSkipEnable)
+		var nextCommandCanExecute =
+			confirmationWordsSourceList
+			.Connect()
+			.WhenValueChanged(x => x.IsConfirmed)
+			.Select(_ => confirmationWordsSourceList.Items.All(x => x.IsConfirmed));
+
+		NextCommand = ReactiveCommand.CreateFromTask(OnNextAsync, nextCommandCanExecute);
+
+		SetSkip();
+
+		confirmationWordsSourceList.AddRange(_words);
+
+		AvailableWords =
+			confirmationWordsSourceList.Items
+									   .Select(x => new RecoveryWordViewModel(x.Index, x.Word))
+									   .OrderBy(_ => Random.Shared.Next())
+									   .ToList();
+
+		var availableWordsSourceList = new SourceList<RecoveryWordViewModel>();
+
+		availableWordsSourceList
+			.DisposeWith(disposables)
+			.Connect()
+			.WhenPropertyChanged(x => x.IsSelected)
+			.Subscribe(x => OnWordSelected(x.Sender));
+
+		availableWordsSourceList.AddRange(AvailableWords);
+
+		SetNextWord();
+
+		var enableCancel = Services.WalletManager.HasWallet();
+		SetupCancel(enableCancel: false, enableCancelOnEscape: enableCancel, enableCancelOnPressed: false);
+	}
+
+	private void SetNextWord()
+	{
+		CurrentWord =
+			CurrentWord?.Index switch
+			{
+				null => ConfirmationWords.First(),
+				int i when i >= 1 && i < ConfirmationWords.Count => ConfirmationWords[i],
+				_ => null
+			};
+
+		if (CurrentWord is { })
 		{
-			SkipCommand = ReactiveCommand.Create(() => NextCommand.Execute(null));
+			EnableAvailableWords(true);
 		}
+	}
 
-		CancelCommand = ReactiveCommand.Create(OnCancel);
+	private void OnWordSelected(RecoveryWordViewModel selectedWord)
+	{
+		if (selectedWord.IsConfirmed)
+		{
+			selectedWord.IsSelected = true;
+			return;
+		}
```

Comment:
I assume if the selection is disabled when a word is confirmed, this part can be removed?